packages/cactus-plugin-persistence-ethereum/src/main/typescript/plugin-persistence-ethereum.ts

```diff
@@ -0,0 +1,960 @@
+
+        if (onError) {
+          try {
+            onError(err);
+          } catch (error: unknown) {
+            this.log.error(
+              "Unexpected error in onError monitor handler:",
+              error,
+            );
+          }
+        }
+      },
+      complete: () => {
+        this.log.info("Watch completed");
+        if (this.watchBlocksSubscription) {
+          this.watchBlocksSubscription.unsubscribe();
+        }
+        this.watchBlocksSubscription = undefined;
+      },
+    });
+  }
+
+  /**
+   * Stop the block monitoring process.
+   * If the monitoring wasn't running - nothing happens.
+   */
+  public stopMonitor(): void {
+    if (this.watchBlocksSubscription) {
+      this.watchBlocksSubscription.unsubscribe();
+      this.watchBlocksSubscription = undefined;
+      this.log.info("stopMonitor(): Done.");
+    }
+  }
+
+  /**
+   * Add new ERC20 token to be monitored by this plugin.
+   * @param address ERC20 contract address.
+   */
+  public async addTokenERC20(address: string): Promise<void> {
+    const checkedAddress = normalizeAddress(address);
+    this.log.info(
+      "Add ERC20 token to monitor changes on it. Address:",
+      checkedAddress,
+    );
+
+    const tokenClient = new TokenClientERC20(this.apiClient, checkedAddress);
+
+    try {
+      await this.dbClient.insertTokenMetadataERC20({
+        address: checkedAddress,
+        name: await tokenClient.name(),
+        symbol: await tokenClient.symbol(),
+        total_supply: parseInt(await tokenClient.totalSupply(), 10),
+      });
+    } catch (err: unknown) {
+      throw new RuntimeError(
+        `Could not store ERC20 token metadata information`,
+        getRuntimeErrorCause(err),
+      );
+    }
+
+    await this.refreshMonitoredTokens();
+  }
+
+  /**
+   * Add new ERC721 token to be monitored by this plugin.
+   * @param address ERC721 contract address.
+   */
+  public async addTokenERC721(address: string): Promise<void> {
+    const checkedAddress = normalizeAddress(address);
+    this.log.info(
+      "Add ERC721 token to monitor changes on it. Address:",
+      checkedAddress,
+    );
+
+    const tokenClient = new TokenClientERC721(this.apiClient, checkedAddress);
+
+    try {
+      await this.dbClient.insertTokenMetadataERC721({
+        address: checkedAddress,
+        name: await tokenClient.name(),
+        symbol: await tokenClient.symbol(),
+      });
+    } catch (err: unknown) {
+      throw new RuntimeError(
+        `Could not store ERC721 token metadata information`,
+        getRuntimeErrorCause(err),
+      );
+    }
+
+    await this.refreshMonitoredTokens();
+  }
+
+  /**
+   * Parse entire block data, detect possible token transfer operations and store the new block data to the database.
+   * Note: token balances are not updated.
+   *
+   * @param block `web3.js` block object.
+   */
+  public async parseBlockData(block: BlockTransactionObject): Promise<void> {
```

Comment:
this function not only parse a block data, but it also inserts the parsed data into a database.
can you rename this function name to e.g. parseAndStoreBlockData to reflect that?

Reply:
@izuru0 Fixed